src/main/java/io/cryostat/recordings/ArchiveRequestGenerator.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright The Cryostat Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.recordings;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+
+import io.cryostat.ws.MessagingServer;
+import io.cryostat.ws.Notification;
+
+import io.vertx.mutiny.core.eventbus.EventBus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ArchiveRequestGenerator {
+
+    private static final String ARCHIVE_RECORDING_SUCCESS = "ArchiveRecordingSuccess";
+    private static final String ARCHIVE_RECORDING_FAIL = "ArchiveRecordingFailed";
+    private final ExecutorService archiveThread = Executors.newCachedThreadPool();
+    private final ExecutorService executor;
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    public ArchiveRequestGenerator(ExecutorService executor) {
+        this.executor = executor;
+    }
+
+    public Future<String> performArchive(
+            ArchiveRequest request, EventBus bus, RecordingHelper recordingHelper) {
```

Comment:
The EventBus and RecordingHelper can be `@Inject`ed into this class, rather than passed via method parameters.

Reply:
Another refactoring idea - rather than picking an ExecutorService explicitly, and then invoking this Generator method directly, I think making this Generator into an `@ApplicationScoped` singleton instance and giving it an `@ConsumeEvent` method would be another less directly coupled approach. This way the execution transfers between worker threads via the EventBus rather than as a direct method call, so it's even more loosely coupled.

Reply:
If I understand it correctly from other places it's used, we'd want to do something like having the ConsumeEvent listen for an Archived Recording notification, but I don't think it would work in this case since the generator needs to produce that notification in the first place, although I suppose we could have the API handler generate a separate notification (e.g. request created) then have the Generator consume it. Is that what you're going for here? Or have I misunderstood how ConsumeEvent is used?

Reply:
Yea, I'm thinking it would be two separate notifications. One would be an internal event bus event that is used for signalling a request to the generator, which would include some payload about what the request is (ie. which target, which recording) as well as the UUID for the job. The callsite (ActiveRecordings endpoint) would generate the UUID and the request event, return the UUID to the client, and once that response has been written out it would fire the request for the generator to start the job. Then the generator would emit an external notification (ie a bus event aimed at the messaging server, to produce a WebSocket event) to notify the client when the job is done with the original UUID.

https://quarkus.io/guides/rest#accessing-context-objects
https://javadoc.io/doc/io.quarkus.resteasy.reactive/resteasy-reactive/latest/org/jboss/resteasy/reactive/server/spi/ServerHttpResponse.html

Very roughly, something like this:

```
UUID recordingReport(ServerHttpResponse resp, ActiveRecording recording) {
  var uuid = UUID.create();
  var req = new GeneratorRequest(uuid, recording);
  resp.addCloseHandler(() -> bus.publish(Generator.class, req));
  return uuid;
}

class Generator {
  @ConsumeEvent(Generator.class)
  void onMessage(GeneratorRequest req) {
    // do processing
    bus.publish(MessagingServer.class, new AsyncJobNotification(uuid));
  }
}
```